### DISCUSSION TITLE
Does create_pydantic_model use optional fields by design

### DISCUSSION THREAD

**Discussion by scott2b** [piccolo-orm/piccolo#665] [Q&A] [answered]
In using a model created with `create_pydantic_model` as a request validator in a web application view, I am finding that the fields are not validated, vs creating my own schema subclassed from BaseModel which will complain, e.g., if the request data is missing fields.

I am trying to determine if I am doing something wrong, or if this is by design.

**Comment by dantownsend** [comment, marked as answer]
There's a parameter you can pass to your columns called `required`. By default it's `False`.  `create_pydantic_model` uses this to determine if a field is optional or not.

```python
class MyTable(Table):
    name = Varchar(required=True)

MyTableModel = create_pydantic_model(MyTable)

# name is now required:
MyTableModel()
>>> Error! 
```

It's not documented well - we need to add a section to this page explaining it:

https://piccolo-orm.readthedocs.io/en/latest/piccolo/serialization/index.html#

**Reply by scott2b** [reply]
Ah, that makes total sense. I should have thought of that. Thank you.

**Reply by dantownsend** [reply]
I've updated the docs now to include this.
